### PR TITLE
nc21

### PR DESCRIPTION
**Pull request by stephdl**
go to nc21

NethServer/dev#6506

### CODE REVIEW

File: root/etc/logrotate.d/rh-mariadb105-nextcloud [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# in the [mysqld] section as follows:
+#
+# [mysqld]
+# log-error=/var/log/rh-mariadb105/nextcloud.log
+
+/var/log/rh-mariadb105/nextcloud.log {
+        create 600 mysql mysql
+        notifempty
+        daily
+        rotate 3
```
Comment (gsanchietti) [inline root/etc/logrotate.d/rh-mariadb105-nextcloud]: Just remove the following options, they will be inherited from the main configuration.
```suggestion
```

File: root/etc/logrotate.d/rh-mariadb105-nextcloud [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        missingok
+        compress
```
Comment (gsanchietti) [inline root/etc/logrotate.d/rh-mariadb105-nextcloud]:
```suggestion
```

**Review by gsanchietti — CHANGES_REQUESTED** [review]
Minor fix

File: createlinks [hunk 3]
```diff
@@ -60,3 +62,12 @@ event_actions("nethserver-nextcloud-save", qw(
...
 ));
+
+# link to backup and restore event
+event_actions("pre-backup-data", qw(
+  nethserver-nextcloud-rh-mariadb105-dump-database 60
+));
+
+event_actions("post-restore-data", qw(
+  nethserver-nextcloud-restore-rh-mariadb105-database 03
+));
```
Comment (gsanchietti) [inline createlinks]: Please add a new line here

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf [hunk 4]
```diff
@@ -8,23 +12,90 @@ function OCC
...
-/usr/bin/mysql --defaults-file=/root/.my.cnf -e "FLUSH PRIVILEGES"
+systemctl is-active --quiet rh-mariadb105-mariadb@nextcloud
+if [[ $? -ne 0 ]]; then
+    systemctl start rh-mariadb105-mariadb@nextcloud
+    max_wait=20
+    wait=0
+    while ! mysql --socket /run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "SHOW DATABASES" 2>/dev/null; do
+        sleep 1
+        wait=$((wait+1))
+        if [ $wait -ge $max_wait ]; then
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: Just to be consistent with other conditions:
```suggestion
        if [[ $wait -ge $max_wait ]]; then
```

File: root/etc/e-smith/events/actions/nethserver-nextcloud-rh-mariadb105-dump-database [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+source /opt/rh/rh-mariadb105/enable
+PATH=$PATH:/opt/rh/rh-mariadb105/root/usr/bin/
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-nextcloud-rh-mariadb105-dump-database]: I'm not sure, we just need to verify it.
Shouldn't be the PATH already prepared from the source command?

Here we do not event export the path.

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf [hunk 6]
```diff
@@ -8,23 +12,90 @@ function OCC
...
+            echo "[ERROR] Can't start rh-mariadb105-mariadb@nextcloud"
+            exit 1
+        fi
+    done
+fi
+
+if [[ -d '/var/lib/mysql/nextcloud' ]]; then
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: Can we add also a query to mysql to see if `nextcloud` db exists?
I fear the directory could not be completely removed if the admin has enabled slow query log or similar.

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf [hunk 7]
```diff
@@ -1,4 +1,8 @@
 #!/bin/bash
+source /opt/rh/rh-mariadb105/enable
+PATH=$PATH:/opt/rh/rh-mariadb105/root/usr/bin/
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: I'm not sure, we just need to verify it.
Shouldn't be the PATH already prepared from the source command?
Is really needed the export below?

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf [hunk 8]
```diff
@@ -8,23 +12,90 @@ function OCC
     TERM=dumb runuser -s /bin/bash apache -c "php -dmemory_limit=512M ./occ $params"
 }
 
+exitOnError () {
```
Comment (gsanchietti) [inline root/etc/e-smith/events/actions/nethserver-nextcloud-conf:12]: When this function is executed, we should get rid also of `tmp_sql`.
Maybe a trap function is better?

**Review by gsanchietti — CHANGES_REQUESTED** [review]

File: root/etc/logrotate.d/rh-mariadb105-nextcloud (hunk 1 above)
Comment (stephdl) [reply root/etc/logrotate.d/rh-mariadb105-nextcloud]: done

File: createlinks (hunk 3 above)
Comment (stephdl) [reply createlinks]: done

File: root/etc/logrotate.d/rh-mariadb105-nextcloud (hunk 2 above)
Comment (stephdl) [reply root/etc/logrotate.d/rh-mariadb105-nextcloud]: done

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf (hunk 4 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: done

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf (hunk 7 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: done

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf (hunk 8 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-conf:12]: done

File: root/etc/e-smith/events/actions/nethserver-nextcloud-rh-mariadb105-dump-database (hunk 5 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-rh-mariadb105-dump-database]: done

File: root/etc/e-smith/events/actions/nethserver-nextcloud-conf (hunk 6 above)
Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: sorry I forgot this one lets go on it

Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-conf]: should be good @gsanchietti

**Review by gsanchietti — APPROVED** [review]
Let's try it!

File: root/etc/e-smith/events/actions/nethserver-nextcloud-restore-rh-mariadb105-database [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+fi
+
+password=`perl -e "use NethServer::Password; print NethServer::Password::store('nextcloud');"`
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "CREATE DATABASE IF NOT EXISTS nextcloud;"
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock --database nextcloud < /var/lib/nethserver/backup/rh-mariadb105/nextcloud.sql
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "ALTER DATABASE nextcloud CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci;"
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "grant all on nextcloud.* to 'nextcloud'@'localhost' identified by '$password';"
+mysql --socket=/run/rh-mariadb105-mariadb/nextcloud-mysql.sock -e "FLUSH PRIVILEGES"
+
+if [[ $? -ne 0 ]]; then
```
Comment (DavidePrincipi) [inline root/etc/e-smith/events/actions/nethserver-nextcloud-restore-rh-mariadb105-database]: This condition checks only the last (FLUSH PRIVILEGES) command exit code. 

- Use  the Bash `set -e` option to automatically exit as soon as an error occurs, OR
- Group together the SQL command strings in a single SQL stream and pipe it to a single  `mysql` command, i.e.:

      ( echo CMD1 ; echo CMD2 ; cat nextcloud.sql ; echo CMD3 ... ) | mysql

Comment (stephdl) [reply root/etc/e-smith/events/actions/nethserver-nextcloud-restore-rh-mariadb105-database]: hello it should be fixed

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**